tools/common/new-release.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2021, Arm Limited
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# invocation example:
+# ./tools/common/new-release.py . 2021 Q1 "02\ :sup:\`nd\` March"
+
+import argparse
+import os
+import re
+
+description = (
+    "Replace dates in ABI files with dates suitable for a new release.")
+
+
+def convert(text, quarter, year, date_of_issue):
+    re_pairs = [
+        (r"(\|release\| replace:: )(.*)", lambda x: x.group(1) + year + quarter),
+        (r"(\|date-of-issue\| replace:: )(.*)", lambda x: x.group(1) + date_of_issue),
+        # In copyright statements:
+        # append this year to last year with a hyphen if it's not a range:
+        # Copyright (c) 2003-2007, 2012, 2018, 2020-2021
+        # becomes:
+        # Copyright (c) 2003-2007, 2012, 2018, 2020-2021
```

Comment:
minor: This is the same as line 35. Has this script already been run over this file!

Reply:
idem: I will fix them both.